ethcore/src/block_queue.rs

```diff
@@ -285,18 +285,24 @@ impl BlockQueue {
 	}
 
 	/// Mark given block and all its children as bad. Stops verification.
-	pub fn mark_as_bad(&mut self, hash: &H256) {
+	pub fn mark_as_bad(&mut self, hashes: &[H256]) {
 		let mut verification_lock = self.verification.lock().unwrap();
+		let mut processing = self.processing.write().unwrap();
```

Comment:
does this introduce a possible deadlock? if so, is it clear and well-documented that this is the canonical order in which to lock these resources?